internal/cloudfront/service.go

```diff
@@ -172,6 +174,66 @@ func newCDNStatus(ings []k8s.CDNIngress, dist Distribution) *v1alpha1.CDNStatus
 	return status
 }
 
+func (s *Service) newDistribution(ingresses []k8s.CDNIngress, group, webACLARN, distARN string) (Distribution, error) {
+	b := NewDistributionBuilder(
+		group,
+		s.Config,
+	)
+
+	var err error
+	var cert certificate.Certificate
+	if s.Config.TLSIsEnabled() {
+		cert, err = s.discoverCert(ingresses)
+		if err != nil {
+			return Distribution{}, fmt.Errorf("discovering TLS cert: %v", err)
+		}
+		b = b.WithTLS(cert.ARN(), s.Config.CloudFrontSecurityPolicy)
+	}
```

Comment:
moved this function from the helpers and diff got messed up, but this part is new.